spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/storage/GcpStorageAutoConfiguration.java

```diff
@@ -56,11 +59,27 @@
 	@ConditionalOnMissingBean
 	public static Storage storage(CredentialsProvider credentialsProvider,
 			GcpStorageProperties gcpStorageProperties,
-			GcpProjectIdProvider projectIdProvider) throws IOException {
+			GcpProjectIdProvider projectIdProvider,
+			@Autowired(required = false) CfConfiguration cfConfiguration) throws IOException {
+		Credentials credentials;
+
+		CredentialsProvider cfCredentialsProvider = null;
+		if (cfConfiguration != null) {
+			cfCredentialsProvider = cfConfiguration.getStorageCredentialsProvider();
```

Comment:
I'm wondering if we could inject `@Qualifier` beans for the credentials with names like "CfStorageCredentials", that would resolve the multiple `Credentials` class of beans and we could just inject them directly and then we could get rid of this if statement

Reply:
I don't think it's worth the trouble, since the storage-specific credentials are only really used by the `Storage` object, so no point exposing another qualified `CredentialsProvider` just to be used in one place.

Also, we would add more complexity around `CredentialsProvider` management where it doesn't exist, nor it needs to. Currently, we know that the `CredentialsProvider` in the context is for the Core credentials. Adding more `CredentialsProvider`s to the context would require qualified usage anywhere we use `CredentialsProvider`, and require that any user knows about how they're managed.